src/main/java/com/minecolonies/api/util/Pond.java

```diff
@@ -16,165 +16,43 @@ public final class Pond
-     *
-     * @param world  World.
-     * @param x      posX.
-     * @param y      posY.
-     * @param z      posZ.
-     * @param vector direction.
-     * @param width  which has to be water.
-     * @param length length has to be water.
-     * @return true if all blocks are water, else false.
-     */
-    private static boolean checkWaterPoolInDirectionXThenZ(
-      @NotNull final BlockGetter world,
-      final int x,
-      final int y,
-      final int z,
-      final int vector,
-      final int width,
-      final int length)
-    {
-        //Check 6 blocks in direction +/- x
-        for (int dx = x - (width / 2 * vector); dx <= x + (width / 2 * vector); dx++)
-        {
-            if (world.getBlockState(new BlockPos(dx, y, z)).getBlock() != Blocks.WATER)
+            if (world.getBlockState(tempPos.set(water.getX(), water.getY() - i, water.getZ())).getBlock() != Blocks.WATER)
             {
                 return false;
             }
         }
-        //Takes the middle x block and searches 3 water blocks to both sides
-        return checkWaterPoolInDirectionZ(world, x + length * vector, y, z, 1) && checkWaterPoolInDirectionZ(world, x + length
-                                                                                                                          * vector, y, z, -1);
-    }
 
-    /**
-     * Checks if all blocks in direction Z are water and if yes from the middle to both sides in direction X all blocks are also water.
-     *
-     * @param world  World.
-     * @param x      posX.
-     * @param y      posY.
-     * @param z      posZ.
-     * @param vector direction.
-     * @param width  which has to be water.
-     * @param length length has to be water.
-     * @return true if all blocks are water, else false.
-     */
-    private static boolean checkWaterPoolInDirectionZThenX(
-      @NotNull final BlockGetter world,
-      final int x,
-      final int y,
-      final int z,
-      final int vector,
-      final int width,
-      final int length)
-    {
-        //Check 6 blocks in direction +/- z
-        for (int dz = z - (width / 2 * vector); dz <= z + (width / 2 * vector); dz++)
+        for (int x = -WATER_POOL_WIDTH_REQUIREMENT / 2; x < WATER_POOL_WIDTH_REQUIREMENT / 2; x++)
         {
-            if (world.getBlockState(new BlockPos(x, y, dz)).getBlock() != Blocks.WATER)
+            for (int z = -WATER_POOL_LENGTH_REQUIREMENT; z < WATER_POOL_LENGTH_REQUIREMENT / 2; z++)
             {
-                return false;
-            }
-        }
-        //Takes the middle z block and searches 3 water blocks to both sides
-        return checkWaterPoolInDirectionX(world, x, y, z + length * vector, 1) && checkWaterPoolInDirectionX(world, x, y, z + length
-                                                                                                                                * vector, -1);
-    }
+                tempPos.set(water.getX() + x, water.getY(), water.getZ() + z);
```

Comment:
It seems kinda unnecessary to use `MutableBlockPos` further down as well.

Also, could use `BlockPos.betweenClosed` or `BlockPos.spiralAround` here.

Reply:
We could use the util for water here?